clang/include/clang/CIR/Dialect/IR/CIRTypeConstraints.td

```diff
@@ -110,4 +110,35 @@ def CIR_AnyFundamentalSIntType
     let cppFunctionName = "isFundamentalSIntType";
 }
 
+//===----------------------------------------------------------------------===//
+// Float Type predicates
+//===----------------------------------------------------------------------===//
+
+def CIR_AnySingleType : CIR_TypeBase<"::cir::SingleType", "single float type">;
```

Comment:
Is there a reason for the explicit global scope?

Reply:
I believe it used to make problems when core mlir dialects where included in out of tree projects/dialects.
Thats why entire mlir has explicit global scope, though I cannot remember the issue it was causing.